core/src/main/java/org/apache/iceberg/encryption/EnvelopeConfig.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.encryption;
+
+import java.io.Serializable;
+import java.util.Objects;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * Configuration for an envelope encryption.
+ */
+public class EnvelopeConfig implements Serializable {
+
+  private final String kekId;
+  private final EncryptionAlgorithm algorithm;
+
+  private EnvelopeConfig(
+          String kekId,
+          EncryptionAlgorithm algorithm) {
+    this.kekId = Preconditions.checkNotNull(kekId,
+            "Cannot construct envelope config because KEK ID is not specified");
+    this.algorithm = Preconditions.checkNotNull(algorithm,
+            "Cannot construct envelope config because encryption algorithm is not specified");
+  }
+
+  public String kekId() {
+    return kekId;
+  }
+
+  public EncryptionAlgorithm algorithm() {
+    return algorithm;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    EnvelopeConfig config = (EnvelopeConfig) o;
+    return  Objects.equals(kekId, config.kekId) && algorithm == config.algorithm;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(kekId, algorithm);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+            .add("kekId", kekId)
+            .add("algorithm", algorithm)
+            .toString();
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * A builder used to create valid {@link EnvelopeConfig}.
+   */
+  public static class Builder {
+    private String kekId;
+    private EncryptionAlgorithm algorithm;
+
+    public Builder singleWrap(String keyId) {
```

Comment:
I think this method especially needs a Java doc or rename. I don't think it is obvious from the methodName what this method is setting in the class and what it will be used for.

Reply:
I'll add a javadoc, "single wrap" and "double wrap" are easier to explain / understand than "key encryption keys" and "master encryption keys".